language-support/hs/bindings/src/DA/Ledger/Valuable.hs

```diff
@@ -0,0 +1,40 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE OverloadedStrings #-}
+
+module DA.Ledger.Valuable(
+    Valuable(..), -- types which can be converted to/from a Ledger API Value
+    ) where
+
+import DA.Ledger.Types
+
+class Valuable a where
+    toValue :: a -> Value
+    fromValue :: Value -> Maybe a
+
+    toRecord :: a -> Record
+    toRecord =
+        Record Nothing
+        . map (\v -> RecordField {label = "", value = v})
+        . (\case VList vs -> vs; v -> [v])
+        . toValue
+
+    fromRecord :: Record -> Maybe a
+    fromRecord =
+        fromValue
+        . VList
+        . map (\RecordField{value} -> value)
```

Comment:
It would be better to label the fields of `RecordField` and use the selector here.

Reply:
The fields of `RecordValue` are labelled. As seen by the use of curly braces in the fragment:
`map (\RecordField{value} -> value)`
I guess you think `map field` would be nicer.
Trouble is that `value` is used as a field name in multiple data records.
So this change is ambiguous.
But we can fix this by changing to a more specific label. `fieldValue`

Reply:
Aside, do you find it slightly confusing that we are discussing Haskell record/field syntax
in an example which is modelling DAML records/fields ? :)